### ISSUE TITLE
Linear Regression Model Comparison Btwn TF, Theano, Caffe, & MXNet

### ISSUE THREAD

**Issue by projectlouis** [apache/incubator-mxnet#5476]
## Background
Currently I'm running some linear regression tests for a non-thesis paper... I'm still naive in the ML realm, so I'm relatively sure I may have misinterpreted something. I'd like to best represent each framework appropriately to ensure a consistent evaluation approach.

I've managed to get a simple example using [airfoil self-noise data](https://archive.ics.uci.edu/ml/datasets/Airfoil+Self-Noise).

The specific data file I use (for my standard pandas read-in format) is at: [airfoil self noise.csv](https://github.com/projectlouis/mlbenchmark/blob/master/airfoil/data/airfoil_self_noise.csv)

Inside the script, you'll notice that all inputs are pushed into a z-score format and this is done across all 4 frameworks. For Tensorflow, since it's iterating and doesn't allow an 'num_epoch' variable, I assumed the following would work:
5000 Epochs --> Iterations = (# Epochs * # Samples) / Batch Size
5000 Epochs --> 58710 Iterations

## Model Used
The Framework Set up is as follows: 
- MLP (3 Layers, 5 Inputs, 1 Output)
- Layer 1 - 25 Hidden, ReLu Activation
- Layer 2 - 5 Hidden, ReLu Activation
- Layer 3 - Final Output -- No Activation
- Batch Size: 128
- Learning Rate: 0.001
- Num Epochs: 5000
- Optimizer: Adam
- Metric: Mean Squared Error 
- Train-Test Split: 80/20

I've highlighted my network creations for each of the four frameworks below:
 [Theano](https://github.com/projectlouis/mlbenchmark/blob/master/airfoil/airfoil_regress_theano.py#L20-L42), 
[TF](https://github.com/projectlouis/mlbenchmark/blob/master/airfoil/airfoil_regress_tf.py#L69-L104), 
[MXNet](https://github.com/projectlouis/mlbenchmark/blob/master/airfoil/airfoil_regress_mxnet.py#L10-L29), 
and [Caffe](https://github.com/projectlouis/mlbenchmark/blob/master/airfoil/airfoil_regress_caffe.py#L101-L113).

## The Issue

**As for the specific question in mind, I'm seeing relatively different R^2 values specifically for MXNet. (Comparing my Predicted versus ideal values inside excel)**

R^2 for MXNet: [0.6463](https://github.com/projectlouis/mlbenchmark/blob/master/airfoil/data/airfoil_regression_mxnet.xlsx)
R^2 for TF: [0.8797](https://github.com/projectlouis/mlbenchmark/blob/master/airfoil/data/airfoil_regression_tf.xlsx)
R^2 for Theano: [0.845](https://github.com/projectlouis/mlbenchmark/blob/master/airfoil/data/airfoil_regression_theano.xlsx)
R^2 for Caffe: [0.5979](https://github.com/projectlouis/mlbenchmark/blob/master/airfoil/data/airfoil_regression_caffe.xlsx)

If anyone could point out issues with the MXNet code, that would be highly appreciative. Caffe seems to act the same as MXNet.... So maybe this isn't an issue?

Cheers,
Louis

## Environment info
Operating System: Windows (Docker Containers)

Package used (Python/R/Scala/Julia): Python

MXNet version: https://hub.docker.com/r/kaixhin/mxnet/

Python version and distribution: Python 2.7.6

**Comment by sxjscience** [comment]
I guess it's due to the initialization method.

**Comment by yajiedesign** [comment]
This issue is closed due to lack of activity in the last 90 days. Feel free to reopen if this is still an active issue. Thanks!
